synapseclient/core/utils.py

```diff
@@ -126,8 +126,8 @@ async def md5_for_file_multiprocessing(
     Returns:
         The MD5 Checksum
     """
-    with tracer.start_as_current_span("Utils::md5_for_file_multiprocessing"):
-        async with md5_semaphore:
+    async with md5_semaphore:
+        with tracer.start_as_current_span("Utils::md5_for_file_multiprocessing"):
```

Comment:
Did you flip these intentionally?

Reply:
Yes, this is intentional. For MD5 calculation I only have 1 additional Python process that gets started up. I found some instability when I let the multi-processing library handle waiting on and scheduling all of these files - Which is why you see the `md5_semaphore` here. By placing the tracer within the `async with md5_semaphore` it means that the trace is only started when this file has gotten the semaphore. I am not capturing the time spent waiting to acquire the semaphore.